transport/internet/splithttp/connection.go

```diff
@@ -1,11 +1,112 @@
 package splithttp
 
 import (
+	"bytes"
+	"errors"
+	"fmt"
 	"io"
 	"net"
+	"strconv"
+	"strings"
 	"time"
 )
 
+var (
+	ErrBadRespCode = errors.New("bad response code")
+	BadCodes       = map[int]struct{}{502: {}, 503: {}, 505: {}}
```

Comment:
everything but 200 OK is bad, can simplify this (also inline the ErrBadRespCode, it carries no type information anyway)

Reply:
Is it possible to have some "OK-like" codes there, like 100, 201, 202, and etc codes that may not require outputing the error? I wasn't sure about this so decided to make "bad codes" list on which we definitely need to return the error

Reply:
the server is xray, so we can define that 200 is the only valid status code here (you can check hub.go to see that this is the only successful status code in practice)